eng/common/scripts/modules/Package-Properties/Package-Properties.psm1

```diff
@@ -0,0 +1,172 @@
+
+            return @{
+                pkgName = $projectJson.name;
+                pkgVersion = $projectJson.version;
+                pkgDirPath = $pkgPath;
+                pkgReadMePath = $pkgReadMePath;
+                pkgChangeLogPath = $pkgChangeLogPath;
+            }
+        }
+    }
+    return $null
+}
+
+function Extract-PythonPkgProps ($pkgPath, $pkgName)
+{
+    $pkgName = $pkgName.Replace('_', '-')
+
+    if (Test-Path (Join-Path $pkgPath "setup.py"))
+    {
+        $setupLocation = $pkgPath.Replace('\','/')
+        pushd $RepoRoot
+        $setupProps = (python -c "import scripts.devops_tasks.common_tasks; obj=scripts.devops_tasks.common_tasks.parse_setup('$setupLocation'); print('{0},{1}'.format(obj[0], obj[1]));") -split ","
+        Write-Host $setupProps
+        popd
+        if (($setupProps -ne $null) -and ($setupProps[0] -eq $pkgName))
+        {
+            $pkgReadMePath = Join-Path $pkgPath "README.md"
+            $pkgReadMePath = If (Test-Path ($pkgReadMePath)) {  $pkgReadMePath } Else { $null }
+            $pkgChangeLogPath = Join-Path $pkgPath "CHANGELOG.md"
+            $pkgChangeLogPath = If (Test-Path ($pkgChangeLogPath)) {  $pkgChangeLogPath } Else { $null }
+
+            return @{
+                pkgName = $setupProps[0];
+                pkgVersion = $setupProps[1];
+                pkgDirPath = $pkgPath;
+                pkgReadMePath = $pkgReadMePath;
+                pkgChangeLogPath = $pkgChangeLogPath;
+            }
+        }
+    }
+    return $null
+}
+
+function Extract-JavaPkgProps ($pkgPath, $pkgName)
+{
+    $projectPath = Join-Path $pkgPath "pom.xml"
+
+    if (Test-Path $projectPath)
+    {
+        $projectData = New-Object -TypeName XML
+        $projectData.load($projectPath)
+        $projectPkgName = Select-XML -Xml $projectData -XPath '/*[local-name()="project"]/*[local-name()="artifactId"]/text()'
+        $pkgVersion = Select-XML -Xml $projectData -XPath '/*[local-name()="project"]/*[local-name()="version"]/text()'
+
+        $pkgReadMePath = Join-Path $pkgPath "README.md"
+        $pkgReadMePath = If (Test-Path ($pkgReadMePath)) {  $pkgReadMePath } Else { $null }
+        $pkgChangeLogPath = Join-Path $pkgPath "CHANGELOG.md"
+        $pkgChangeLogPath = If (Test-Path ($pkgChangeLogPath)) {  $pkgChangeLogPath } Else { $null }
+        
+        if ($projectPkgName.ToString() -eq $pkgName)
+        {
+            return @{
+                pkgName = $pkgName;
+                pkgVersion = $pkgVersion.ToString();
+                pkgDirPath = $pkgPath;
+                pkgReadMePath = $pkgReadMePath;
+                pkgChangeLogPath = $pkgChangeLogPath;
+            }
+        }
+    }
+    return $null
+}
+
+# Takes package name and service Name
+# Returns important properties of the package as related to the language repo
+# Returns a HashTable with Keys @ { pkgName, pkgVersion, pkgDirPath, pkgReadMePath, pkgChangeLogPath }
+function Get-PkgProperties {
```

Comment:
It would be awesome to add a List-Packages function that would provide this information for all the packages in the repo and/or service directory.